components/backup/src/writer.rs

```diff
@@ -190,6 +199,76 @@ impl BackupWriter {
     }
 }
 
+/// A writer writes Raw kv into SST files.
+pub struct BackupRawKVWriter {
+    name: String,
+    cf: String,
+    write: Writer,
+    limiter: Option<Arc<RocksIOLimiter>>,
+}
+
+impl BackupRawKVWriter {
+    /// Create a new BackupRawKVWriter.
+    pub fn new(
+        db: Arc<DB>,
+        name: &str,
+        cf: &str,
+        limiter: Option<Arc<RocksIOLimiter>>,
+    ) -> Result<BackupRawKVWriter> {
+        let write = RocksSstWriterBuilder::new()
+            .set_in_memory(true)
+            .set_cf(super::endpoint::rawkv_cf(cf)?)
+            .set_db(RocksEngine::from_ref(&db))
+            .build(name)?;
+        Ok(BackupRawKVWriter {
+            name: name.to_owned(),
+            cf: cf.to_owned(),
+            write: Writer::new(write),
+            limiter,
+        })
+    }
+
+    /// Write Kv_pair to buffered SST files.
+    pub fn write(&mut self, kv_pairs: &[Result<KvPair>], need_checksum: bool) -> Result<()> {
```

Comment:
This is ok though, just an suggestion, you can try to make this function more like `BackupWriter::write`. Consider:
```rust
    pub fn write<I>(&mut self, entries: I, need_checksum: bool) -> Result<()>
    where
        I: Iterator<Item = KvPair>,
```
Then you can write `writer.write(batch.drain(..), need_checksum)` and don't need to invoke `batch.clear()`.

Reply:
try it